jvm/src/main/scala/uk/gov/homeoffice/drt/db/dao/StaffShiftsDao.scala

```diff
@@ -0,0 +1,38 @@
+package uk.gov.homeoffice.drt.db.dao
+
+import slick.jdbc.PostgresProfile.api._
+import slick.lifted.TableQuery
+import uk.gov.homeoffice.drt.db.tables.{StaffShiftRow, StaffShiftsTable}
+import scala.concurrent.Future
+
+trait IStaffShiftsDao {
+  def insertOrUpdate(staffShiftRow: StaffShiftRow): Future[Int]
+
+  def getStaffShiftsByPort(port: String): Future[Seq[StaffShiftRow]]
+
+  def getStaffShiftsByPortAndTerminal(port: String, terminal: String): Future[Seq[StaffShiftRow]]
+
+  def getStaffShiftByPortAndTerminalAndShiftName(port: String, terminal: String, shiftName: String): Future[Seq[StaffShiftRow]]
+
+  def deleteStaffShift(port: String, terminal: String, shiftName: String): Future[Int]
+}
+
+case class StaffShiftsDao(db: Database) extends IStaffShiftsDao {
+  val staffShiftsTable: TableQuery[StaffShiftsTable] = TableQuery[StaffShiftsTable]
+
+  override def insertOrUpdate(staffShiftRow: StaffShiftRow): Future[Int] =
+    db.run(staffShiftsTable.insertOrUpdate(staffShiftRow))
+
+  override def getStaffShiftsByPort(port: String): Future[Seq[StaffShiftRow]] =
+    db.run(staffShiftsTable.filter(_.port === port).result)
+
+  override def getStaffShiftsByPortAndTerminal(port: String, terminal: String): Future[Seq[StaffShiftRow]] =
+    db.run(staffShiftsTable.filter(s => s.port === port && s.terminal === terminal).result)
+
+  override def getStaffShiftByPortAndTerminalAndShiftName(port: String, terminal: String, shiftName: String): Future[Seq[StaffShiftRow]] =
```

Comment:
Since the primary key is (port, term, name) this can only return 0 or 1 result, so it should be an Option rather than a Seq